Jenkinsfile

```diff
@@ -319,18 +318,36 @@ def job = {
                     sh "docker push ${config.dockerRegistry}${dockerRepo}:master-latest"
 
                     if (config.release) {
-                        sh "docker tag ${config.dockerRegistry}${dockerRepo}:${config.docker_tag} ${config.dockerRegistry}${dockerRepo}:${config.docker_tag}-${env.BUILD_NUMBER}"
-                        sh "docker push ${config.dockerRegistry}${dockerRepo}:${config.docker_tag}-${env.BUILD_NUMBER}"
+                        // docker_tag for release builds does not include the build number
+                        // here we add it back so an image version with the build number always exists
+                        sh "docker tag ${config.dockerRegistry}${dockerRepo}:${config.docker_tag} ${config.dockerRegistry}${dockerRepo}:${config.docker_tag}-rc${env.BUILD_NUMBER}"
+                        sh "docker push ${config.dockerRegistry}${dockerRepo}:${config.docker_tag}-rc${env.BUILD_NUMBER}"
                     }
                 }
             }
         }
     }
+
+    // Build CCloud ksqlDB image
+
+    if (!config.release && !config.isPrJob) {
+        stage('Trigger CCloud ksqlDB Docker image job') {
+            build job: "confluentinc/cc-docker-ksql/master",
+                wait: false,
+                parameters: [
+                    booleanParam(name: "NIGHTLY_BUILD", value: true),
+                    string(name: "NIGHTLY_BUILD_NUMBER", value: "${env.BUILD_NUMBER}"),
+                    string(name: "CP_BETA_VERSION", value: "${config.cp_version}"),
+                    string(name: "CP_BETA_BUILD_NUMBER", value: "${config.packaging_build_number}"),
+                    string(name: "KSQLDB_ARTIFACT_VERSION", value: "${config.ksql_db_artifact_version}")
+                ]
+        }
+    }
 }
 
 def post = {
     withDockerServer([uri: dockerHost()]) {
-        repos = config.dockerArtifacts + config.dockerRepos
+        repos = config.dockerArtifacts + config.dockerRepos + config.dockerPullDeps
```

Comment:
Added this so dependency images are cleaned up in addition to the built ones. The script doesn't pull dependencies in `PROMOTE_TO_PRODUCTION` mode and the dependency repo doesn't exist in the production registry, but I think that should be fine? As in, trying to clean up a registry + repo combination that doesn't exist shouldn't cause the script to error. Please correct me if I'm mistaken.

Reply:
@vcrfxia Apologies, but I got that wrong and I mentioned some more detail in the other PR. You only want to delete the images you build.

Reply:
No worries, reverted the cleanup logic change.